external/avro/src/main/scala/org/apache/spark/sql/avro/AvroFileFormat.scala

```diff
@@ -153,6 +154,31 @@ private[sql] class AvroFileFormat extends FileFormat
   }
 
   override def supportDataType(dataType: DataType): Boolean = AvroUtils.supportsDataType(dataType)
+
+  override def checkFieldNames(names: Seq[String]): Unit = {
+    names.foreach(checkFieldName)
+  }
+
+  private def checkFieldName(name: String): Unit = {
+    val length = name.length
+    if (length == 0) {
+      throw new SchemaParseException("Empty name")
+    } else {
+      val first = name.charAt(0)
+      if (!Character.isLetter(first) && first != '_') {
+        throw new SchemaParseException("Illegal initial character: " + name)
+      } else {
+        var i = 1
+        while (i < length) {
+          val c = name.charAt(i)
+          if (!Character.isLetterOrDigit(c) && c != '_') {
+            throw new SchemaParseException("Illegal character in: " + name)
+          }
+          i += 1
+        }
+      }
```

Comment:
Looks like this code is directly copied from `Schema.validateName`. At minimum, we need to provide a copyright attribution here. It's a shame that the function isn't exposed publicly.

Rather than duplicating the validation logic here, I'm wondering if we should work around it by creating a placeholder schema with the fields from `checkFieldNames` (the types can be arbitrary) and asking Avro to parse it, and treating the field names as valid if Avro doesn't throw a `SchemaParseException`, e.g.:
```
        val emptyBuilder = SchemaBuilder.record("foo").fields()
        val fullBuilder = names.foldLeft(emptyBuilder)((builder, name) => builder.requiredInt(name))
        fullBuilder.endRecord()
```